pymc4/flow/transformed_executor.py

```diff
@@ -17,124 +26,112 @@ def validate_state(self, state):
         """Validate that the model is not in a bad state."""
         return
 
-    def modify_distribution(self, dist, model_info, state):
+    def modify_distribution(
+        self, dist: ModelType, model_info: Mapping[str, Any], state: SamplingState
+    ) -> ModelType:
         """Apply transformations to a distribution."""
         dist = super().modify_distribution(dist, model_info, state)
         if not isinstance(dist, distribution.Distribution):
             return dist
-        scoped_name = scopes.variable_name(dist.name)
 
-        if dist.transform is None or dist.model_info.get(  # do nothing else if no transform is set
-            "autotransformed", False
-        ):  # already autotransformed, do nothing else
-            return dist
-        transform = dist.transform
-        transformed_scoped_name = scopes.variable_name(
-            # double underscore stands for transform
-            "__{}_{}".format(transform.name, dist.name)
+        return transform_dist_if_necessary(dist, state, allow_transformed_and_untransformed=True)
```

Comment:
here is where most of the change happened -- I factored this logic into 3 functions, with this one switch, and had both executors use it.